spec/cdn_source_spec.rb

```diff
@@ -239,38 +195,33 @@ def typhoeus_non_http_response_future(code)
       end
 
       it 'retries after unexpected HTTP error' do
-        @source.expects(:download_typhoeus_impl_async).
-          with_url('http://localhost:4321/all_pods_versions_2_0_9.txt').
-          at_most(5).
-          returns(typhoeus_http_response_future(503)).
-          returns(typhoeus_http_response_future(503)).
-          returns(typhoeus_http_response_future(503)).
-          returns(typhoeus_http_response_future(503)).
-          then.
-          returns(typhoeus_http_response_future(200, {}, 'BeaconKit/1.0.0'))
-        @source.expects(:download_typhoeus_impl_async).
-          with_url('http://localhost:4321/Specs/2/0/9/BeaconKit/1.0.0/BeaconKit.podspec.json').
-          returns(typhoeus_http_response_future(200, {}, ''))
-
-        [4, 8, 16, 32].each do |seconds|
-          @source.expects(:sleep_async).with(seconds).returns(resolved_event)
+        WebMock.stub_request(:get, 'http://localhost:4321/all_pods_versions_2_0_9.txt').
+          to_return(:status => 503, :headers => {}, :body => '').
+          to_return(:status => 503, :headers => {}, :body => '').
+          to_return(:status => 503, :headers => {}, :body => '').
+          to_return(:status => 503, :headers => {}, :body => '').
+          to_return(:status => 200, :headers => {}, :body => 'BeaconKit/1.0.0')
+
+        WebMock.stub_request(:get, 'http://localhost:4321/Specs/2/0/9/BeaconKit/1.0.0/BeaconKit.podspec.json').
+          to_return(:status => 200, :headers => {}, :body => '')
+
+        [4, 8, 16, 32].each_with_index do |seconds, i|
+          @source.expects(:backoff_time).with(Pod::CDNSource::MAX_NUMBER_OF_RETRIES - i).returns(seconds)
```

Comment:
@jasl perhaps change back to the way it was?
```ruby
  [4, 8, 16, 32].each do |seconds|		
           @source.expects(:sleep_async).with(seconds)		
  end
```

Reply:
good, simplier than current one